auto_gptq/nn_modules/qlinear/qlinear_cuda_old.py

```diff
@@ -266,10 +266,10 @@ def forward(self, x):
             weight = (scales * (weight - zeros))
             weight = weight.reshape(weight.shape[0] * weight.shape[1], weight.shape[2])
 
-            out = torch.matmul(x.half(), weight)
+            out = torch.matmul(x.to(weight.dtype), weight)
         out = out.half().reshape(out_shape)
         out = out + self.bias if self.bias is not None else out
-        return out
+        return out.to(x.dtype)
```

Comment:
This introduces a bug when using the CUDA kernel in fp32.

Reply:
What kind of bug? Can you please explain?

Reply:
When `use_cuda_fp16=False`, there is a cast `x = x.to(torch.float32)` which results in the output dtype being wrong with the change above. This is fixed in https://github.com/PanQiWei/AutoGPTQ/pull/382.

Reply:
Thanks @fxmarty!